daml-lf/testing-tools/src/main/scala/com/digitalasset/daml/lf/engine/testing/SemanticTester.scala

```diff
@@ -218,11 +224,31 @@ class SemanticTester(
                       s"Expected exercise event $comparedScenarioExerciseEvent but got $ledgerExerciseEventToCompare"
                     )
                   }
-                  // add the exercise children to the stack
                   state.copy(
-                    remainingLedgerEventIds = ledgerExerciseEvent.children ++: remainingLedgerEventIds,
-                    remainingScenarioNodeIds = scenarioExercisesNode.children ++: remainingScenarioNodeIds,
+                    remainingScenarioNodeIds = remainingScenarioNodeIds,
+                    remainingLedgerEventIds = remainingLedgerEventIds,
                   )
+
+                case scenarioExercisesNode: NodeExercises[
+                      L.NodeId,
+                      AbsoluteContractId,
+                      Tx.Value[AbsoluteContractId]] if !exerciseAddedChildren =>
+                  val (ledgerExerciseEvent, _) =
```

Comment:
make this a bit more explicit by immediately getting the `._1.children` out

Reply:
Done